linux/drivers/hid/hid-magicmouse.c

```diff
@@ -566,10 +568,13 @@ static int magicmouse_setup_input(struct input_dev *input, struct hid_device *hd
 		 * loaded, whether connected through bluetooth or USB.
 		 */
 		if (hdev->vendor == BT_VENDOR_ID_APPLE) {
-			if (input->id.version == TRACKPAD2_2021_BT_VERSION)
+			if (input->id.version == TRACKPAD2_2021_BT_VERSION) {
+				input->name = "Apple Inc. Magic Trackpad 2021";
+			} else if (input->id.version == TRACKPAD_2024_BT_VERSION) {
+				input->name = "Apple Inc. Magic Trackpad USB-C";
```

Comment:
If I remember correctly ,the name reported by the trackpad usn-c over USB is "Apple Inc. Magic Trackpad." , I'm not sure if the device name reported via Bluetooth needs to reflect the current hardware version. However, I tried to submit a patch.
https://patchwork.kernel.org/project/linux-input/patch/20241202060545.21193-1-soragoto@soragoto.io/

Reply:
@mr-cal and @dev-soragoto should the name be "Apple Inc. Magic Trackpad"? And rename the vars to `TRACKPAD2_BT_VERSION` and `TRACKPAD2_USBC_BT_VERSION`?

Reply:
For mine libinput shows for Bluetooth

```
Device:           Apple Inc. Magic Trackpad USB-C
Kernel:           /dev/input/event30
Group:            10
Seat:             seat0, default
Size:             162x115mm
Capabilities:     pointer gesture
Tap-to-click:     disabled
Tap-and-drag:     enabled
Tap drag lock:    disabled
Left-handed:      disabled
Nat.scrolling:    enabled
Middle emulation: disabled
Calibration:      n/a
Scroll methods:   *two-finger edge
Click methods:    button-areas *clickfinger
Disable-w-typing: n/a
Disable-w-trackpointing: n/a
Accel profiles:   flat *adaptive custom
Rotation:         n/a
```

For USB-C it shows

```
Device:           Apple Inc. Magic Trackpad
Kernel:           /dev/input/event23
Group:            7
Seat:             seat0, default
Size:             162x115mm
Capabilities:     pointer gesture
Tap-to-click:     disabled
Tap-and-drag:     enabled
Tap drag lock:    disabled
Left-handed:      disabled
Nat.scrolling:    enabled
Middle emulation: disabled
Calibration:      n/a
Scroll methods:   *two-finger edge
Click methods:    button-areas *clickfinger
Disable-w-typing: enabled
Disable-w-trackpointing: enabled
Accel profiles:   flat *adaptive custom
Rotation:         n/a
```

Reply:
Here's my context, I could be misunderstanding the commit history: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/drivers/hid/hid-magicmouse.c?h=v6.13-rc2&id=5768701edcb7d7ff8abaa085996a5c1cb30d765e

Reply:
> Here's my context, I could be misunderstanding the commit history: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/drivers/hid/hid-magicmouse.c?h=v6.13-rc2&id=5768701edcb7d7ff8abaa085996a5c1cb30d765e

I understand your point. To keep the device name consistent with the USB connection, could we make the changes this way

```c
if (hdev->vendor == BT_VENDOR_ID_APPLE) {
	switch (input->id.version) {
		case TRACKPAD2_2021_BT_VERSION:
		case TRACKPAD_2024_BT_VERSION;
			input->name = "Apple Inc. Magic Trackpad";
			break;
		default:
			input->name = "Apple Inc. Magic Trackpad2";
			break;
	}
} else { /* USB_VENDOR_ID_APPLE */
	input->name = hdev->name;
}
```

However, there is still a question, why is the default device name's suffix 2, Before 2021, were the device names reported via USB all like this?

Reply:
> Here's my context, I could be misunderstanding the commit history: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/drivers/hid/hid-magicmouse.c?h=v6.13-rc2&id=5768701edcb7d7ff8abaa085996a5c1cb30d765e

I understand your point. To keep the device name consistent with the USB connection, could we make the changes this way

```c
if (hdev->vendor == BT_VENDOR_ID_APPLE) {
	switch (input->id.version) {
		case TRACKPAD2_2021_BT_VERSION:
		case TRACKPAD_2024_BT_VERSION;
			input->name = "Apple Inc. Magic Trackpad";
			break;
		default:
			input->name = "Apple Inc. Magic Trackpad2";
			break;
	}
} else { /* USB_VENDOR_ID_APPLE */
	input->name = hdev->name;
}
```

However, there is still a question, why is the default device name's suffix 2, Before 2021, were the device names reported via USB all like this?

Reply:
Is https://github.com/mr-cal/Linux-Magic-Trackpad-2-USB-C-Driver/blob/2a495f025d7ea838a3396b3e556e3f5d2d1160c6/linux/drivers/hid/hid-magicmouse.c#L567-L576 how you would prefer it?

I am quite confused by the suffix of 2. My Lightning has suffix 2. My USBC does not.

Reply:
When the lightning version is now connected via usb its without the suffix.

Reply:
> Is
> 
> https://github.com/mr-cal/Linux-Magic-Trackpad-2-USB-C-Driver/blob/2a495f025d7ea838a3396b3e556e3f5d2d1160c6/linux/drivers/hid/hid-magicmouse.c#L567-L576
> 
> how you would prefer it?
> I am quite confused by the suffix of 2. My Lightning has suffix 2. My USBC does not.

I suspect that only a few versions of the hardware will have the suffix 2,  but now the default device name's suffix 2, I don't know if doing this is correct.

Reply:
Since it seems like only the 2021 trackpad reports the name with a "2" when connected via USB, then it is the only model that needs special handling.  Maybe we should to just leave this section as-is?

Reply:
> Since it seems like only the 2021 trackpad reports the name with a "2" when connected via USB, then it is the only model that needs special handling. Maybe we should to just leave this section as-is?

Do you mean it this way?

```
 if (hdev->vendor == BT_VENDOR_ID_APPLE) { 
 	if (input->id.version == TRACKPAD2_2021_BT_VERSION) { 
 		input->name = "Apple Inc. Magic Trackpad 2"; 
 	} else { 
 		input->name = "Apple Inc. Magic Trackpad"; 
 	} 
 } else { /* USB_VENDOR_ID_APPLE */ 
 	input->name = hdev->name; 
 } 
```

Reply:
> Since it seems like only the 2021 trackpad reports the name with a "2" when connected via USB, then it is the only model that needs special handling. Maybe we should to just leave this section as-is?

https://github.com/mr-cal/Linux-Magic-Trackpad-2-USB-C-Driver/blob/7bf74255ea0dd6a9c0c7790cf4c115b52637e976/linux/drivers/hid/hid-magicmouse.c#L568-L572

From this piece of code, it seems that the 2021 version's name probably doesn't include "2" either. At the same time, the 2024 version still doesn’t include "2" either.

Reply:
@mr-cal my Ligthning Version has PPN A1535 and has 613 (0x265) which is not `#define TRACKPAD2_2021_BT_VERSION 0x110`

Reply:
@mr-cal @dev-soragoto do you know how you find out the Version Number of `TRACKPAD2_2021_BT_VERSION 0x110` and `TRACKPAD_USBC_BT_VERSION 0x314` via Bluetooth. I can not help myself finding this out for my bluetooth Trackpad. I want to make sure, my Lightning Trackpad indeed has the 0x110 Version, since it shows `input->name = "Apple Inc. Magic Trackpad 2";` when connected via Bluetooth.

Reply:
I've been finding it with:
```
sudo dmesg | grep "magic"
[ 5951.519896] magicmouse 0005:004C:0324.0003: input,hidraw2: BLUETOOTH HID v3.14 Mouse [Callahan’s Magic Trackpad] on 0c:54:15:d5:31:93
```

Mine is reporting a custom name after I connected it to a macbook for comparison, but I believe the `v3.14` in that string is represented by `0x314`.

Reply:
Alright, this was my output of `sudo dmesg | grep "magic"`:
```
> sudo dmesg | grep "magic"
[sudo] password for admin: 
[  597.114036] hid_magicmouse: loading out-of-tree module taints kernel.
[  597.224027] magicmouse 0005:004C:0265.0008: unknown main item tag 0x0
[  597.224621] magicmouse 0005:004C:0265.0008: input,hidraw3: BLUETOOTH HID v3.11 Mouse [Magic Trackpad von Sebastian] on 7c:21:4a:24:2f:24
```
Added now `#define TRACKPAD2_A1535_2733_VERSION 0x311` to the list of item that do not have a 2 in the suffix.  See https://github.com/mr-cal/Linux-Magic-Trackpad-2-USB-C-Driver/pull/3/commits/8d294fc0f8b77f6eb95c87fb33252a114377e06f